src/main/java/hudson/remoting/Channel.java

```diff
@@ -270,7 +274,9 @@
      * without telling us anything, the {@link SocketOutputStream#write(int)} will
      * return right away, and the socket only really times out after 10s of minutes.
      */
-    private volatile long lastHeard;
+    @GuardedBy("lastHeardLock")
+    private long lastHeard;
+    private Object lastHeardLock = new String("lastHeardLock");
```

Comment:
How is locking on a String a bad idea?
Its not like I am locking on an interned String which is a very bad idea :) 
infact locking on a String helps when debugging as rather than seeing a random reference you have a good visual clue as to what Locks are held.

in any case will change to a Lock as should at least be less overhead.